framework/src/media/MediaRecorderImpl.cpp

```diff
@@ -280,31 +280,35 @@ recorder_result_t MediaRecorderImpl::stop()
 	return RECORDER_OK;
 }
 
-void MediaRecorderImpl::stopRecorder(bool completed)
+void MediaRecorderImpl::stopRecorder(bool normalMode)
```

Comment:
like a playback, please add error handling code, I mean code related to xrun.
It looks so good.